connectd/connectd.c

```diff
@@ -2154,12 +2182,20 @@ int main(int argc, char *argv[])
 {
 	struct daemon *daemon;
 	bool developer;
+	struct stat finfo;
+	int sodium_fd;
 
 	setup_locale();
 
 	/* Common subdaemon setup code. */
 	developer = subdaemon_setup(argc, argv);
 
+	/* Check if lib-sodium is using fd 5 and it looks like /dev/random */
+	sodium_fd = -1;
+	if(!isatty(5) && errno == ENOTTY && 0 == fstat(5, &finfo))
+		if (S_ISCHR(finfo.st_mode) && !finfo.st_size)
+			sodium_fd = 5;
+
```

Comment:
Does this work?  libsodium doesn't have an explicit init call, so I gathered it initialized on first use?  If so, a call to pseudorand(1) would work.

But anyway, this is a weird nasty hack.  We should try to detect a char device in dev_report_fd and allow a single one, on the theory it's libsodium.

Reply:
> Does this work? libsodium doesn't have an explicit init call, so I gathered it initialized on first use? If so, a call to pseudorand(1) would work.

It caught it on my machine(tm). I'll switch it over to allowing one char fd in `dev_report_fd` instead 👍